### PR TITLE
better file equality test, same as file-equal-p

### LINKED ISSUE #13: TOC navigation: nov-visit-relative-file: Couldn’t locate document
Perhaps I'm missing something, but I can't navigate to a location from the TOC with <kbd>Enter</kbd>.

Debugger entered--Lisp error: (error "Couldn’t locate document")
  signal(error ("Couldn’t locate document"))
  error("Couldn't locate document")
  nov-visit-relative-file("Text/cover.html" nil)
  apply(nov-visit-relative-file ("Text/cover.html" nil))
  nov-browse-url(13)
  funcall-interactively(nov-browse-url 13)
  call-interactively(nov-browse-url nil nil)
  command-execute(nov-browse-url)

### PR DESCRIPTION
**Pull request by rolandwalker**
`file-equal-p` is only available in Emacs 24+

fixes #13, at least for me

### CODE REVIEW

File: nov.el [hunk 1]
```diff
@@ -510,7 +510,8 @@ the HTML is rendered with `shr-render-region'."
          (directory (file-name-directory current-path))
          (path (file-truename (nov-make-path directory filename)))
          (index (nov-find-document (lambda (doc)
-                                     (string-suffix-p path (cdr doc))))))
+                                     (equal (file-attributes (file-truename (cdr doc)))
+                                            (file-attributes (file-truename path)))))))
```
Comment (wasamasa) [inline nov.el]: No need to use `file-truename` again (it has been used a few lines above already).

**Comment by wasamasa** [comment]
Generally, before merging this I'd like to understand why it must be identical by file attributes and why normalization is necessary not only for the resolved relative path (which is a combination of the current file's path and the link target that may contain `..`), but also for each document file (the names of which are in `nov-documents`.  Could it be that `temporary-file-directory` is weird on macOS and is a symlink to another place?  If yes, that would explain why you'd need to use `file-truename` on both, but not why you'd also look up `file-attributes`.

**Comment by rolandwalker** [comment]
[`file-equal-p` source](https://github.com/emacs-mirror/emacs/blob/af11532bb5f5ff8229b84c3ab6888069764bbddf/lisp/files.el#L5440)

To determine that two files are the same file, it is necessary and sufficient for the last two elements of the return value of `file-attributes` to be equal: inode number and filesystem device number.  Comparing the full list rather than the last two elements is just cleaner to write, and will always be equivalent.

Yes, the purpose of `file-truename` is to resolve symlinks.  I think there are some platforms/situations for which `file-attributes` will return attributes for the symlink itself, rather than the underlying inode.

Yes, macOS is weird: `$TMPDIR` and `temporary-file-directory` point to `/var/folders/…/…/T/`, where `/var` is a symlink to `/private/var`.

Per your advice, I removed `file-truename` from both elements, though I think that only one of them may have been truenamed already, which may indeed have been why the suffix test could return a false negative.

**Comment by wasamasa** [comment]
One is, the other wasn't.  I fixed the code so that both are.

**Pull request closed without merging**